samcli/commands/remote_invoke/invoke/cli.py

```diff
@@ -0,0 +1,146 @@
+        stack_name,
+        resource_id,
+        event,
+        event_file,
+        output_format,
+        parameter,
+        ctx.region,
+        ctx.profile,
+        config_file,
+        config_env,
+    )
+
+
+def do_cli(
+    stack_name: str,
+    resource_id: str,
+    event: str,
+    event_file: TextIOWrapper,
+    output_format: str,
+    parameter: dict,
+    region: str,
+    profile: str,
+    config_file: str,
+    config_env: str,
+) -> None:
+    """
+    Implementation of the ``cli`` method
+    """
+    from samcli.commands.exceptions import UserException
+    from samcli.commands.remote_invoke.remote_invoke_context import RemoteInvokeContext
+    from samcli.lib.remote_invoke.exceptions import (
+        ErrorBotoApiCallException,
+        InvalideBotoResponseException,
+        InvalidResourceBotoParameterException,
+    )
+    from samcli.lib.remote_invoke.remote_invoke_executors import RemoteInvokeExecutionInfo, RemoteInvokeOutputFormat
+    from samcli.lib.utils.boto_utils import get_boto_client_provider_with_config, get_boto_resource_provider_with_config
+
+    boto_client_provider = get_boto_client_provider_with_config(region_name=region)
+    boto_resource_provider = get_boto_resource_provider_with_config(region_name=region)
+
+    try:
+        with RemoteInvokeContext(
+            boto_client_provider=boto_client_provider,
+            boto_resource_provider=boto_resource_provider,
+            stack_name=stack_name,
+            resource_id=resource_id,
+        ) as remote_invoke_context:
+
+            if not resource_id and not stack_name:
+                LOG.error("Atleast 1 of --stack-name or --resource-id parameters should be provided")
+                return
+
+            output_format_enum_value = RemoteInvokeOutputFormat.DEFAULT
+            if output_format == "full-boto-response":
+                output_format_enum_value = RemoteInvokeOutputFormat.FULL_BOTO_RESPONSE
+
+            # if no event nor event_file arguments are given, read from stdin
+            if not event and not event_file:
+                LOG.info("Neither --event nor --event-file options have been provided, reading from stdin")
+                event_file = cast(TextIOWrapper, sys.stdin)
+
+            remote_invoke_input = RemoteInvokeExecutionInfo(
+                payload=event, payload_file=event_file, parameters=parameter, output_format=output_format_enum_value
+            )
+
+            remote_invoke_result = remote_invoke_context.run(remote_invoke_input=remote_invoke_input)
+
+            if remote_invoke_result.is_succeeded():
+                if remote_invoke_result.log_output:
+                    remote_invoke_context.stderr.write(remote_invoke_result.log_output.encode())
+                output_response = cast(str, remote_invoke_result.response)
+                remote_invoke_context.stdout.write(output_response.encode())
+            else:
+                raise cast(Exception, remote_invoke_result.exception)
```

Comment:
Should we raise whatever is been raised from the call, or should we wrap it into specific user exception and raise that instead?

Reply:
So any exception that occurs in `/lib` is set to `remote_invoke_result.exception` parameter. Line 144 raises that exception which is caught on line [145](https://github.com/aws/aws-sam-cli/blob/08f296126e2d3d1793bea36056e6db31712283f6/samcli/commands/remote_invoke/invoke/cli.py#L145) and raised as `UserException`.  Do you mean we should name the specific UserExceptions being raised in line 145?

Reply:
Oh, if we are already sorting that out in the inner layers, than we should be good for now. Thanks for explaining it.